arch/arm/src/common/arm_createstack.c

```diff
@@ -237,19 +246,27 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
  ****************************************************************************/
 
 #ifdef CONFIG_STACK_COLORATION
-void arm_stack_color(FAR void *stackbase, size_t nbytes)
```

Comment:
By changing the name, we are loosing information. It would be best to leave it the same or document the parameters to all the functions.

Reply:
Then I'd like to remane the the size parameter of do_stackcheck to nbytes as well.
What do you think about moving arm_stack_color to arm_checkstack.c? It is already a public function declared in arm_internal.h, and it suits there better.

Reply:
> Then I'd like to rename the the size parameter of do_stackcheck to nbytes as well.

Sure! Less ambiguity is  always better.

> What do you think about moving arm_stack_color to arm_checkstack.c?

I have not looked at the files in a while: To give you perspective on why it might have been separate, (but we should ask @patacongo, he has a far better memory then I.) In the past NuttX maintained high scalability by using a "one function per file" methodology and Kconfig to do the garbage prevention, as apposed to linker garbage collection. 
 
In this case I think the answer, answered by the the question: Is there any functionality in arm_checkstack that is not dependent on stack coloring?  If so they can be roommates.